src/twisted/internet/defer.py

```diff
@@ -2022,8 +2024,8 @@ def _inlineCallbacks(
             appCodeTrace = traceback.tb_next
             assert appCodeTrace is not None
 
-            if _PYPY:
-                # PyPy as of 3.7 adds an extra frame.
+            if _STACK_LEVEL_INCOMPATIBLE:
+                # PyPy as of 3.7 and before 7.3.14 adds an extra frame.
```

Comment:
Maybe something like this, with a comment that in the future this can be removed.

```suggestion
            _old_pypy_stack_compatibility = _PYPY and implementation.version < (7, 3, 14)
            if _old_pypy_stack_compatibility:
                # PyPy before 7.3.14 adds an extra frame.
                # This code can be removed once we no longer need to support PyPy 7.3.13 or older.
```

Reply:
Sure, these changes look fine to me. I will let the twisted maintainers commit these suggestions.